File: cpp/leetcode/无重复字符的最长子字串/1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        charSet = set()
        max_length = 0
        left = 0

        for right in range(len(s)):
            while s[right] in charSet:
                charSet.remove(s[left])
                left += 1
            charSet.add(s[right])
            max_length = max(max_length, right - left + 1)
        return max_length

if __name__ == "__main__":
    solution = Solution()
    s = "pwwkew"
    print(f"{solution.lengthOfLongestSubstring(s)}")

File: cpp/leetcode/无重复字符的最长子字串/1.cpp
#include <iostream>
#include <string>
#include <unordered_set>
#include <algorithm>

using namespace std;

class Solution {
public:
    int lengthOfLongestSubstring(string& s) {
        unordered_set<char> charSet;
        int left = 0;
        int max_length = 0;

        for (int right = 0; right < s.length(); ++right) {
            while (charSet.find(s[right]) != charSet.end()) {
                charSet.erase(s[left]);
                ++left;
            }
            charSet.insert(s[right]);
            max_length = max(max_length, right - left + 1);
        }
        return max_length;
    }
};

int main() {
    Solution solution;
    string s = "pwwkew";
    cout << solution.lengthOfLongestSubstring(s) << endl;
}